install/nvidia.sh

```diff
@@ -0,0 +1,98 @@
+# ==============================================================================
+# Hyprland NVIDIA Setup Script for Arch Linux
+# ==============================================================================
+# This script automates the installation and configuration of NVIDIA drivers
+# for use with Hyprland on Arch Linux, following the official Hyprland wiki.
+#
+# Author: https://github.com/Kn0ax
+# 
+# ==============================================================================
+
+# --- GPU Detection ---
+# Check if nvidia gpu exists, exit silently if not found
+gpu_info=$(lspci | grep -i 'nvidia')
+if [ -z "$gpu_info" ]; then
+    exit 0
+fi
+
+# --- Colors for better readability ---
+C_RESET='\033[0m'
+C_RED='\033[0;31m'
+C_YELLOW='\033[0;33m'
+C_BOLD='\033[1m'
+
+# --- Driver Selection ---
+# Turing (16xx, 20xx), Ampere (30xx), Ada (40xx), and newer recommend the open-source kernel modules
+if echo "$gpu_info" | grep -q -E "RTX [2-9][0-9]|GTX 16"; then
+    NVIDIA_DRIVER_PACKAGE="nvidia-open-dkms"
+else
+    NVIDIA_DRIVER_PACKAGE="nvidia-dkms"
+fi
+
+# Check which kernel is installed and set appropriate headers package
+KERNEL_HEADERS="linux-headers"  # Default
+if pacman -Q linux-zen &>/dev/null; then
+    KERNEL_HEADERS="linux-zen-headers"
+elif pacman -Q linux-lts &>/dev/null; then
+    KERNEL_HEADERS="linux-lts-headers"
+elif pacman -Q linux-hardened &>/dev/null; then
+    KERNEL_HEADERS="linux-hardened-headers"
+fi
+
+# Enable multilib repository for 32-bit libraries
+if ! grep -q "^\[multilib\]" /etc/pacman.conf; then
+    sudo sed -i '/^#\[multilib\]/,/^#Include/ s/^#//' /etc/pacman.conf
+fi
+
+# Install packages
+PACKAGES_TO_INSTALL=(
+    "${KERNEL_HEADERS}"    
+    "${NVIDIA_DRIVER_PACKAGE}"
+    "nvidia-utils"
+    "lib32-nvidia-utils"
+    "egl-wayland"
+    "libva-nvidia-driver" # For VA-API hardware acceleration
+    "qt5-wayland"
+    "qt6-wayland"
+)
+
+if ! yay -Syu --needed --noconfirm "${PACKAGES_TO_INSTALL[@]}"; then
```

Comment:
Don't think we need to be defensive about this. `--needed` already ensures we're only installing if we need to. We should expect that yay works otherwise the whole thing is dead anyway. So let's just drop the defensive testing.